src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+    verifyLoggedUser(USER_NICKNAME_HELPER, "log-out-link");
+    backToHome();
+    ifLaggingThenRefresh();
+    verifyLoggedUser(USER_NICKNAME_HELPER, "login-logout");
+    helpOut();
+    goToUserPage();
+    ifLaggingThenRefresh();
+    goToOfferHelp();
+    verifyOfferHelpTask();
+  }
+
+  @Test
+  /** Tests functionality of having a helper mark a task as complete */
+  public void _06_Userpage_AsLoggedHelper_CompleteTask() {
+    completeTaskAsHelper();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+
+    // Refreshes page due to flakiness of test caused by the partial refresh of the div in the page.
+    // Without refreshing this often resulted in a a stale element error
+    driver.navigate().refresh();
+    goToOfferHelp();
+
+    // Location of first task listed in the complete task table in userpage
+    String taskCompletedXPath = "//tbody[@id='complete-task-body']/tr[1]";
+
+    String taskOverview =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskCompletedXPath + "/td[1]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("overview"), taskOverview);
+
+    String taskStatus =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskCompletedXPath + "/td[2]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("status"), taskStatus);
+
+    String taskNeighbor =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskCompletedXPath + "/td[3]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("nickname"), taskNeighbor);
+
+    String taskPoints =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskCompletedXPath + "/td[4]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("points"), taskPoints);
+
+    verifyTaskDetails(taskCompletedXPath + "/td[1]");
+  }
+
+  @Test
+  /** Tests functionality of neighbor user verifying that helper did indeed complete a task */
+  public void _07_Userpage_AsLoggedUser_VerifyCompletedTask() {
+    // Logs out by using 'logout-href' element id;
+    logOut("logout-href");
+    loginUser(USER_EMAIL);
+    ifLaggingThenRefresh();
+
+    // Location of first task listed in the await verification table
+    String awaitVerifTaskXPath = "//tbody[@id='await-verif-body']/tr[1]";
+
+    String taskOverview =
```

Comment:
As the suggestion above for function `_06_Userpage_AsLoggedHelper_CompleteTask()`, using a foor loop here can be great.